src/utils/enumSchema.ts

```diff
@@ -0,0 +1,9 @@
+import * as yup from 'yup';
+import {validationErrors} from '@app/constants/validationErrors.constant';
+
+export function enumSchema<T>(enumObject: Record<string, T>): yup.NumberSchema {
+  return yup.number().oneOf(
+    Object.values(enumObject).map((value) => value as number),
+    validationErrors.INVALID_VALUE
```

Comment:
What will be if you pass an `enum` key? `Object.values(enumObject)` returns also keys if I remember correctly.

Reply:
You are right. It will work because I use ```as number``` here, but ```Object.values(enumObject)``` returns also keys. In case I try to pass enum key like "SPORT" I am getting a correct 400 error and a message is: ""categoryId must be a `number` type, but the final value was: `NaN` (cast from the value `\"SPORT\"`)." So it works, but technically it is not the best way.

Reply:
> You are right. It will work, but `Object.values(enumObject)` returns also keys. In case I try to pass enum key like "SPORT" I am getting a correct 400 error and a message is: ""categoryId must be a `number` type, but the final value was: `NaN` (cast from the value `\"SPORT\"`)." So it works, but technically it is not the best way. After all I can extract only values if I will use filter before map. In this case error will be the same (about NaN), but array will consists only from values without keys.
> 
> ```
> Object.values(enumObject)
>       .filter((val) => typeof val === 'number')
>       .map((value) => value as number)
> ```

Reply:
And one more. What I said on a meeting. When I use another approach like 
``` categoryId: yup.mixed<CategoriesEnum>().oneOf(Object.values(CategoriesEnum)), ```
then when I pass enum key as SPORT I will have not 400 error, but 500. And when I pass ordinary string (not enum key) it will work correctly and return 400.

Reply:
Thanks, in my opinion it will be more professional only with corresponding values in the array.